symbolic-debuginfo/src/macho.rs

```diff
@@ -65,15 +65,49 @@ impl<'d> MachObject<'d> {
             .map_err(MachError::new)
     }
 
-    /// Parses and loads the BCSymbolMap into the object.
+    /// Parses and loads the [`BCSymbolMap`] into the object.
     ///
-    /// The symbolmap must match the object, there is nothing in the symbol map which allows
-    /// this call to verify this.
+    /// The bitcode symbol map must match the object, there is nothing in the symbol map
+    /// which allows this call to verify this.
     ///
     /// Once the symbolmap is loaded this object will transparently resolve any hidden
     /// symbols using the provided symbolmap.
-    pub fn load_bc_symbol_map(&mut self, bc_symbol_map: BCSymbolMap<'d>) {
-        self.bcsymbolmap = Some(bc_symbol_map);
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use symbolic_debuginfo::bcsymbolmap::BCSymbolMap;
+    /// use symbolic_debuginfo::macho::MachObject;
+    ///
+    /// let object_data =
+    ///     std::fs::read("tests/fixtures/2d10c42f-591d-3265-b147-78ba0868073f.dwarf-hidden")
+    ///         .unwrap();
```

Comment:
This example is great! It could be confusing to see the "test/fixtures..." path in a doc comment, though, so what if we hide that line (and same for the symbolmap below)?

```suggestion
    /// # let object_data =
    /// #     std::fs::read("tests/fixtures/2d10c42f-591d-3265-b147-78ba0868073f.dwarf-hidden")
    /// #         .unwrap();
```

Reply:
Hmm, it would make the `object_data` variable come out of nowhere with no idea on what its type would be.  I prefer this as it's fully self contained, I don't think it's a big leap to infer that's just a path name.  Plus anyone who knows doctests, which will be most people reading this, will know exactly what's going on.

Reply:
I could be convinced to do something ugly like:
```
/// let object_data = std::fs::read("some/path/object.dwarf");
/// # let object_data = <the full thing that actually works>
```
I guess, if you really prefer.  But I don't see it as much better to be honest.

Reply:
Turns out, in the standard library they annotate such examples with `norun` and simply ensure it compiles, since there will be tests covering this anyway. Given that this is quite an odd path to read, we could do that, too? 

See [`std::fs::File` source](https://doc.rust-lang.org/stable/src/std/fs.rs.html#65-78).

Reply:
I would like to keep the example as a test-case too.  I've gone with some compromise where I fake-read the data from a path like it appears in an .xcarchive in a comment while hiding where we actually load it from.